### PR TITLE
Media Identifier

### PR DESCRIPTION
**Pull request by brianfleming**
## Description

While working on upcoming designs for additional elements in media cards I took the the opportunity to do some space and style optimizations mostly focused on the media cards. This PR includes:

-  a new component `<MediaIdentifier...` styled after other media card elements such as `<LastRequestDate...` in order to provide consistent styles
- decouple the setting of font size and color for `<MediaTypeDisplayIcon...` because this was causing colors to be set incorrectly for some usage. Instead the size and color are set now in css files to match our other patterns
- Update the `<MediaSlug...` to include `<MediaIdentifier...` instead of having it set separately, so that the `<MediaIdentifier...` can be shown along side other media card metadata such as request count
- include copy to clipboard for `<MediaIdentifier...` so that users can click the button and get the identifier to use in searching which is the primary use of the slug
- Use `media_slug` wherever the `<MediaIdentifier...` is output with a fallback to title because of the presence of imported fact-checks that still appear in lists since they do not have a slug
- Use CSS to set the ellipsis overflow for some URLs instead of relying on char count
- introduce a size for thumbnails so we can reduce the size of the small media card and get more media on the screen at one time
- sort some prop types as necessary
- improve some css for compact bullet separator
- Add the same title/quote/description that is seen in the media list to the dialog

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

manual and sandbox

### AFTER - the `<MediaIdentifier...` button shows a tooltip with the type of media and the full slug in case its cut off in the main UI
Screenshot 2024-11-20 at 2 06 42 PM

### AFTER - clicking the `<MediaIdentifier...` button copies to clipboard then outputs a toast message, this was copied from the API key copy to clipboard feature
Screenshot 2024-11-20 at 11 42 43 AM

### AFTER - If a URL is output in a small media card its given more room before being cutoff via ...
Screenshot 2024-11-20 at 2 07 36 PM

### AFTER. - Dialog title
Screenshot 2024-11-20 at 3 48 31 PM

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)
- [x] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed

### CODE REVIEW

**Comment by brianfleming** [comment]
@sarahkeh can you take a first review on this? You touched that import/export dialog recently and there is a integration test failure that I can't figure out what I changed to make it fail, so if I could get your help with that during that review process that would be appreciated 🙌 

Also, @caiosba offered to do a final review after yours since this PR touches a lot of different places and he has worked on them recently, also the media identifier slug that is now copyable via this PR is part of a fix he is working on for the search

**Comment by caiosba** [comment]
@sarahkeh ping me here when you finish your review, please!

**Comment by brianfleming** [comment]
> @brianfleming I think that's one of the flaky tests, I've seen that one fail on me before. Usually when I rerun it, it passes. Let me take a closer look though.

that was my first hope @sarahkeh !
the hope of "it couldn't have possibly been my screw up! it must be flaky?!?"
but then I got less confident, because I ran it a few times in a row with that same failure, so 🤷

**Comment by sarahkeh** [comment]
Just checked a Build Failure email I got yesterday for the Tab component patch, the tests were:
1) integration (chrome) web should search by keywords
2) integration (chrome) web should import and export items

When I reran the tests, it passed. A quick glance over at your code and I don't see anything that would cause those tests to fail consistently, and the class that needs to be selected is still there.  What does seem consistent is that these are two tests that contain a verbose_wait specifically for Elasticsearch, so I think that may be the underlying cause.  

@caiosba let me know if you have other thoughts, but that's my assessment of these failing tests.

**Comment by caiosba** [comment]
@sarahkeh , yeah, the integration tests are more flaky since we moved from Travis to GitHub Actions... as we discussed in the last Frontend Team call, I think we'll need to dedicate some time to work on making them more stable. But I think that @brianfleming was referencing a unit test... can you please look into that?

**Comment by brianfleming** [comment]
> Details

I think it was a unit test I was referring to @caiosba this one:
`web should import and export items`
right?

thats the one that I saw fail consistently when I ran smoke tests against this branch, which made me think it was something I messed up, but @sarahkeh thinks its flakiness she has seen before? is that right @sarahkeh ?

**Comment by caiosba** [comment]
Ahh OK... I'm going to talk to @sarahkeh about it now.

**Comment by caiosba** [comment]
@sarahkeh , as we discussed now, ping me here when you finish your review, please!

File: src/app/components/cds/media-cards/MediaIdentifier.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  const swallowClick = (e) => {
+    e.stopPropagation();
+  };
+
+  const handleCopyToClipboard = () => {
+    setFlashMessage((
+      <FormattedMessage
+        defaultMessage="The media identifier has been copied to the clipboard"
+        description="success message fro when the identifier slug for this piece of media has been copied to the clipboard"
```
[the remark this answers is not in the document]
Comment (brianfleming) [reply src/app/components/cds/media-cards/MediaIdentifier.js]: ha, so funny, just went to fix this and found that there were 2 others of the same typo. Looks like we all like copy-paste!

**Review by sarahkeh — APPROVED** [review]
Tested on my end, looks good!

**Comment by sarahkeh** [comment]
@caiosba all yours now!

File: src/app/components/cds/media-cards/MediaIdentifier.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  theme: PropTypes.string,
+  tooltip: PropTypes.bool,
+  variant: PropTypes.string,
+};
+
+MediaIdentifier.defaultProps = {
+  variant: 'contained',
+  theme: 'lightBeige',
+  tooltip: true,
+};
```
Comment (caiosba) [inline src/app/components/cds/media-cards/MediaIdentifier.js:91]: Usually we declare the `defaultProps` before the `propTypes`, but this doesn't make any difference.

File: src/app/components/cds/media-cards/MediaIdentifier.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      : buttonContent
+  );
+};
+
+MediaIdentifier.propTypes = {
+  mediaType: PropTypes.string.isRequired,
+  slug: PropTypes.any.isRequired,
+  theme: PropTypes.string,
+  tooltip: PropTypes.bool,
+  variant: PropTypes.string,
```
Comment (caiosba) [inline src/app/components/cds/media-cards/MediaIdentifier.js:84]: Please declare the `intl` prop here as well (e.g.: `intl: intlShape.isRequired` - need to import `intlShape` from `react-intl`).

**Review by caiosba — APPROVED** [review]
Great stuff @brianfleming , good refactoring for the `<MediaIndentifier />` component. I also tested it locally and works well. Please add the `intl` `propType` before merging.

File: src/app/components/cds/media-cards/MediaIdentifier.js (hunk 2 above)
Comment (brianfleming) [reply src/app/components/cds/media-cards/MediaIdentifier.js]: oh yeah, I will keep an eye out of this in the future too. resolved here!

File: src/app/components/cds/media-cards/MediaIdentifier.js (hunk 3 above)
Comment (brianfleming) [reply src/app/components/cds/media-cards/MediaIdentifier.js:91]: consider yourself declared!

**Comment by brianfleming** [comment]
nice! thanks for the great reviews @sarahkeh and @caiosba 🙌

**Pull request merged**